src/puppetlabs/puppetdb/scf/storage.clj

```diff
@@ -167,34 +168,35 @@
-   :catalog-hash-miss  (timer [ns-str "default" "catalog-hash-miss-time"])
-   :replace-catalog    (timer [ns-str "default" "replace-catalog-time"])
-
-   :gc                 (timer [ns-str "default" "gc-time"])
-   :gc-catalogs        (timer [ns-str "default" "gc-catalogs-time"])
-   :gc-params          (timer [ns-str "default" "gc-params-time"])
-   :gc-environments    (timer [ns-str "default" "gc-environments-time"])
-   :gc-report-statuses (timer [ns-str "default" "gc-report-statuses"])
-   :gc-fact-paths  (timer  [ns-str "default" "gc-fact-paths"])
-
-   :updated-catalog    (counter [ns-str "default" "new-catalogs"])
-   :duplicate-catalog  (counter [ns-str "default" "duplicate-catalogs"])
-   :duplicate-pct      (gauge [ns-str "default" "duplicate-pct"]
-                              (let [dupes (value (:duplicate-catalog performance-metrics))
-                                    new   (value (:updated-catalog performance-metrics))]
-                                (float (kitchensink/quotient dupes (+ dupes new)))))
-   :catalog-volatility (histogram [ns-str "default" "catalog-volitilty"])
-
-   :replace-facts     (timer [ns-str "default" "replace-facts-time"])
-
-   :store-report      (timer [ns-str "default" "store-report-time"])
+   :add-resources      (timer metrics-registry [ns-str "default" "add-resources"])
+   :add-edges          (timer metrics-registry [ns-str "default" "add-edges"])
+
+   :resource-hashes    (timer metrics-registry [ns-str "default" "resource-hashes"])
+   :catalog-hash       (timer metrics-registry [ns-str "default" "catalog-hash"])
+   :add-new-catalog    (timer metrics-registry [ns-str "default" "new-catalog-time"])
+   :catalog-hash-match (timer metrics-registry [ns-str "default" "catalog-hash-match-time"])
+   :catalog-hash-miss  (timer metrics-registry [ns-str "default" "catalog-hash-miss-time"])
+   :replace-catalog    (timer metrics-registry [ns-str "default" "replace-catalog-time"])
+
+   :gc                 (timer metrics-registry [ns-str "default" "gc-time"])
+   :gc-catalogs        (timer metrics-registry [ns-str "default" "gc-catalogs-time"])
+   :gc-params          (timer metrics-registry [ns-str "default" "gc-params-time"])
+   :gc-environments    (timer metrics-registry [ns-str "default" "gc-environments-time"])
+   :gc-report-statuses (timer metrics-registry [ns-str "default" "gc-report-statuses"])
+   :gc-fact-paths  (timer metrics-registry [ns-str "default" "gc-fact-paths"])
+
+   :updated-catalog    (counter metrics-registry [ns-str "default" "new-catalogs"])
+   :duplicate-catalog  (counter metrics-registry [ns-str "default" "duplicate-catalogs"])
+   :duplicate-pct      (gauge-fn metrics-registry [ns-str "default" "duplicate-pct"]
+                                 (fn []
+                                   (let [dupes (value (:duplicate-catalog performance-metrics))
+                                         new   (value (:updated-catalog performance-metrics))]
```

Comment:
I just would like to note that this self-referential code bothered me. :smile: When I updated the library the first error I got was from this causing a `NullPointerException`.